### PR TITLE
Add Tuition and Fees Deduction

### LINKED ISSUE #4035: Tuition and fees deduction
IRS Form 8917

Data available in the PUF (~$7 billion)

### PR DESCRIPTION
**Pull request by pxu12**
Fixes #4035

### CODE REVIEW

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees_deduction/joint.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees_deduction/joint.yaml:1]: we can name the folder `tuition_and_fees`

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees_deduction/other.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees_deduction/other.yaml:1]: can you rename this to non_joint.yaml

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees_deduction/joint.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+brackets:
+  - threshold:
+      2002-01-01: 0
+      2004-01-01: 0
+    amount:
+      2002-01-01: 3_000
+      2004-01-01: 4_000
+  - threshold:
+      2002-01-01: 130_000
+      2004-01-01: 130_000
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees_deduction/joint.yaml:11]:
```suggestion
```

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees_deduction/joint.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      2002-01-01: 0
+metadata:
+  threshold_unit: /1
+  amount_unit: currency_USD
+  amount_period: year
+  type: single_amount
+  reference:
+    - title: 2020 Form 8917
+      href: https://www.irs.gov/pub/irs-pdf/f8917.pdf#page=2
+    - title: I.R.C. § 222(b)(2)(B)(ii)
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees_deduction/joint.yaml:28]:
```suggestion
    # Law was repealed in 2021 
    - title: I.R.C. § 222(b)(2)(B)(ii)
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/joint.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+description: Modified adjusted gross income limit of tuition and fees deduction for joint filers.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/joint.yaml:1]:
```suggestion
description: The IRS limits the tuition and fees deduction to joint filers to this amount, based on adjusted gross income.
```

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/joint.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      2004-01-01: 2_000
+  - threshold:
+      2002-01-01: .inf
+      2004-01-01: 160_000
+    amount:
+      2002-01-01: 0
+metadata:
+  threshold_unit: currency_USD
+  amount_unit: currency_USD
+  amount_period: year
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/joint.yaml:21]:
```suggestion
  period: year
```

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/joint.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  type: single_amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/joint.yaml:22]: add label

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/non_joint.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      2004-01-01: 2_000
+  - threshold:
+      2002-01-01: .inf
+      2004-01-01: 80_000
+    amount:
+      2002-01-01: 0
+metadata:
+  threshold_unit: currency_USD
+  amount_unit: currency_USD
+  amount_period: year
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/non_joint.yaml:21]:
```suggestion
  period: year
```

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/non_joint.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  type: single_amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/non_joint.yaml:22]: add label

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/non_joint.yaml [hunk 8]
```diff
@@ -0,0 +1,28 @@
+description: Modified adjusted gross income limit of tuition and fees deduction for non-joint filers.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/non_joint.yaml:1]:
```suggestion
description: The IRS limits the tuition and fees deduction to non-joint filers to this amount, based on adjusted gross income
```

File: policyengine_us/variables/gov/irs/income/taxable_income/deductions/tuition_and_fees_deduction.py [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    definition_period = YEAR
+
+    def formula(tax_unit, period, parameters):
+        qualified_tuition_expenses = add(
+            tax_unit, period, ["qualified_tuition_expenses"]
+        )
+        adjusted_gross_income = tax_unit("adjusted_gross_income", period)
+        filing_status = tax_unit("filing_status", period)
+        p = parameters(period).gov.irs.deductions.tuition_and_fees
+        limit = select(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/irs/income/taxable_income/deductions/tuition_and_fees_deduction.py:18]: can we use a where statement here instead?

Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/irs/income/taxable_income/deductions/tuition_and_fees_deduction.py:18]:
```suggestion
        cap = select(
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/irs/income/taxable_income/deductions/tuition_and_fees_deduction.py [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+    def formula(tax_unit, period, parameters):
+        qualified_tuition_expenses = add(
+            tax_unit, period, ["qualified_tuition_expenses"]
+        )
+        adjusted_gross_income = tax_unit("adjusted_gross_income", period)
+        filing_status = tax_unit("filing_status", period)
+        p = parameters(period).gov.irs.deductions.tuition_and_fees
+        cap = where(
+            filing_status == filing_status.possible_values.JOINT,
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/irs/income/taxable_income/deductions/tuition_and_fees_deduction.py:19]:
```suggestion
        joint = filing_status == filing_status.possible_values.JOINT
        cap = where(
            joint,
```

File: policyengine_us/variables/gov/irs/income/taxable_income/deductions/tuition_and_fees_deduction.py [hunk 11]
```diff
@@ -0,0 +1,23 @@
+from policyengine_us.model_api import *
+
+
+class tuition_and_fees_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Tuition and fees deduction"
+    unit = USD
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/irs/income/taxable_income/deductions/tuition_and_fees_deduction.py:8]: add reference

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/non_joint.yaml [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+      2002-01-01: .inf
+      2004-01-01: 80_000
+    amount:
+      2002-01-01: 0
+metadata:
+  threshold_unit: currency_USD
+  amount_unit: currency_USD
+  period: year
+  type: single_amount
+  label: IRS tuition and fees deduction cap to non-joint filers
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/non_joint.yaml:23]:
```suggestion
  label: IRS tuition and fees deduction non-joint filers cap 
```

File: policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/joint.yaml [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+      2002-01-01: .inf
+      2004-01-01: 160_000
+    amount:
+      2002-01-01: 0
+metadata:
+  threshold_unit: currency_USD
+  amount_unit: currency_USD
+  period: year
+  type: single_amount
+  label: IRS tuition and fees deduction cap to joint filers
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/tuition_and_fees/joint.yaml:23]:
```suggestion
  label: IRS tuition and fees deduction joint filers cap 
```

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: we need to add this to the list of IRS deductions

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Comment by pxu12** [comment]
Hi @PavelMakarchuk, regarding the deduction list, in the tax form, this deduction is in the adjustment to income section:
Screen Shot 2024-03-07 at 3 00 48 PM
I searched online and it's said to be an above-the-line deduction:
Screen Shot 2024-03-07 at 3 07 31 PM
I assumed that means that the deduction can be claimed both when itemizing and not itemizing, so I added it to both lists. But I'm not sure that's correct.

File: policyengine_us/parameters/gov/irs/deductions/deductions_if_itemizing.yaml [hunk 14]
```diff
@@ -8,6 +8,7 @@ values:
     - casualty_loss_deduction
     - qualified_business_income_deduction
     - wagering_losses_deduction
+    - tuition_and_fees_deduction
 metadata:
   unit: variable
   name: taxable_income_deductions_if_itemizing
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/deductions_if_itemizing.yaml:14]:
```suggestion
  name: taxable_income_deductions_if_itemizing
```

File: policyengine_us/parameters/gov/irs/deductions/deductions_if_itemizing.yaml [hunk 15]
```diff
@@ -8,6 +8,7 @@ values:
     - casualty_loss_deduction
     - qualified_business_income_deduction
     - wagering_losses_deduction
+    - tuition_and_fees_deduction
 metadata:
   unit: variable
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/deductions_if_itemizing.yaml:13]:
```suggestion
  unit: list
  period: year
```

File: policyengine_us/parameters/gov/irs/deductions/deductions_if_not_itemizing.yaml [hunk 16]
```diff
@@ -3,6 +3,7 @@ values:
   2018-01-01:
     - standard_deduction
     - qualified_business_income_deduction
+    - tuition_and_fees_deduction
 metadata:
   unit: variable
   name: taxable_income_deductions_if_itemizing
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/deductions_if_not_itemizing.yaml:9]:
```suggestion
```

File: policyengine_us/parameters/gov/irs/deductions/deductions_if_not_itemizing.yaml [hunk 17]
```diff
@@ -3,6 +3,7 @@ values:
   2018-01-01:
     - standard_deduction
     - qualified_business_income_deduction
+    - tuition_and_fees_deduction
 metadata:
   unit: variable
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/deductions_if_not_itemizing.yaml:8]:
```suggestion
  unit: list
  period: year
```

File: policyengine_us/parameters/gov/irs/deductions/deductions_if_itemizing.yaml [hunk 18]
```diff
@@ -8,6 +8,7 @@ values:
     - casualty_loss_deduction
     - qualified_business_income_deduction
     - wagering_losses_deduction
+    - tuition_and_fees_deduction
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/irs/deductions/deductions_if_itemizing.yaml:11]: make a 2021-01-01: 
  2021-01-01:
    - charitable_deduction
    - interest_deduction
    - salt_deduction
    - medical_expense_deduction
    - casualty_loss_deduction
    - qualified_business_income_deduction
    - wagering_losses_deduction

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/irs/deductions/deductions_if_itemizing.yaml:11]: same for not itemized to show that it was repealed

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**